### PR TITLE
Added watchOS Support

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-ios/103)

### CODE REVIEW

File: Sources/Classes/RSDeviceInfo.m [hunk 1]
```diff
@@ -9,23 +9,46 @@
...
+        _manufacturer = @"Apple";
         _attTrackingStatus = RSATTNotDetermined;
+        _type = [self getDeviceType];
     }
+
     return self;
 }
 
+- (NSString *) getDeviceType {
+#if TARGET_OS_WATCH
```
Comment (pallabmaiti) [inline Sources/Classes/RSDeviceInfo.m:42]: Where is iPadOS?

File: Sources/Classes/Public/RSDeviceInfo.h [hunk 2]
```diff
@@ -12,6 +12,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface RSDeviceInfo : NSObject
 
+- (NSString *) getDeviceType;
```
Comment (pallabmaiti) [inline Sources/Classes/Public/RSDeviceInfo.h:15]: If this method only required for this class. Don't need to expose as public. And make it static method.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]
Commented.

File: Sources/Classes/RSDeviceInfo.m (hunk 1 above)
Comment (desusai7) [reply Sources/Classes/RSDeviceInfo.m]: Actually, we cannot check if the app is running on iPadOS at compile time. We don't have something like `TARGET_OS_IPAD`. 

So setting `type` the same way we are setting `name` in RSDeviceInfo.m

File: Sources/Classes/Public/RSDeviceInfo.h (hunk 2 above)
Comment (desusai7) [reply Sources/Classes/Public/RSDeviceInfo.h:15]: Okay, got it. 
Also removed this method in the latest commit as we are setting context.device.type programatically

**Review by pallabmaiti — CHANGES_REQUESTED** [review]
Update the version please.

File: Rudder.podspec [hunk 3]
```diff
@@ -9,9 +9,11 @@ Pod::Spec.new do |s|
   s.homepage         = "https://github.com/rudderlabs/rudder-sdk-ios"
   s.license          = { :type => "Apache", :file => "LICENSE" }
   s.author           = { "Rudderstack" => "arnab@rudderlabs.com" }
-  s.source           = { :git => "https://github.com/rudderlabs/rudder-sdk-ios.git", :tag => "v1.2.2" }
+  s.source           = { :git => "https://github.com/rudderlabs/rudder-sdk-ios.git", :tag => "v1.3.0" }
```
Comment (pallabmaiti) [inline Rudder.podspec:12]: Can you use here from `v1.3.0` to `v#{s.version}`

**Review by pallabmaiti — COMMENTED** [review]
Can you update the project version as well?

File: Rudder.podspec (hunk 3 above)
Comment (desusai7) [reply Rudder.podspec:12]: Yes

File: Sources/Classes/RSEventRepository.m [hunk 4]
```diff
@@ -10,9 +10,12 @@
 #import "RSElementCache.h"
 #import "RSUtils.h"
 #import "RSLogger.h"
+
+#import "WKInterfaceController+RSScreen.h"
 #import "UIViewController+RSScreen.h"
 
 static RSEventRepository* _instance;
+static BOOL firstForeGround = YES;
```
Comment (pallabmaiti) [inline Sources/Classes/RSEventRepository.m:18]: Don't use static variable. Use class variable.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]
Please do the necessary change.

File: Sources/Classes/RSEventRepository.m (hunk 4 above)
Comment (desusai7) [reply Sources/Classes/RSEventRepository.m]: Okay, changed accordingly

**Comment by desusai7** [comment]
@pallabmaiti , made the changes accordingly

File: Sources/Classes/RSEventRepository.m [hunk 5]
```diff
@@ -481,30 +485,49 @@ - (void) __checkApplicationUpdateStatus {
...
     }
     NSString *previousVersion = [preferenceManager getBuildVersionCode];
-    
     NSString *currentVersion = [[NSBundle mainBundle] infoDictionary][@"CFBundleShortVersionString"];
     
     if (!previousVersion) {
         [[RSClient sharedInstance] track:@"Application Installed" properties:@{
             @"version": currentVersion
         }];
+        [preferenceManager saveBuildVersionCode:currentVersion];
```
Comment (pallabmaiti) [inline Sources/Classes/RSEventRepository.m:530]: I think previous implementation is correct.

Comment (desusai7) [reply Sources/Classes/RSEventRepository.m:530]: Actually, we need to save the version of the app in the user defaults, either when the app is installed or when the app got updated, as it is created or changed in those two scenarios itself. 

But in the previous implementation, the only problem is we are overriding the version in the user defaults every time the app is opened even though it is not needed. 

So I had changed it to this way to decrease the unnecessary writes to the user defaults.

Does it make sense (or) shall we move to the previous implementation?

Comment (pallabmaiti) [reply Sources/Classes/RSEventRepository.m:530]: Okay Understand.

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**